api/lib/domain/services/user-reconciliation-service.js

```diff
@@ -11,13 +14,37 @@ function findMatchingCandidateIdForGivenUser(matchingUserCandidates, user) {
 
   return _(['firstName', 'middleName', 'thirdName'])
     .map(_findCandidatesMatchingWithUser(standardizedMatchingUserCandidates, standardizedUser))
-    .filter(containsOneElement)
+    .filter(_containsOneElement)
     .flatten()
     .map('id')
     .first() || null;
 }
 
-function containsOneElement(arr) {
+async function findMatchingOrganizationStudentIdForGivenUser({ organizationId, user: { firstName, lastName, birthdate }, studentRepository }) {
+  const students = await studentRepository.findByOrganizationIdAndUserBirthdate({
+    organizationId,
+    birthdate,
+  });
+
+  if (students.length === 0) {
+    throw new NotFoundError('There were no students matching');
```

Comment:
There were no students matching with organisation and birthdate ?

Reply:
Prio basse